### PR TITLE
Agenda Item Endpoint

### PR DESCRIPTION
**Pull request by mjoyce91**
Use with https://github.com/MetaPhase-Consulting/mock-fsbid/pull/245

### CODE REVIEW

File: talentmap_api/settings.py [hunk 1]
```diff
@@ -542,6 +542,7 @@ def skip_lb_requests(record):
 HRDATA_URL_EXTERNAL = get_delineated_environment_variable('HRDATA_URL_EXTERNAL', 'http://mock_fsbid:3333/HR')
 AVATAR_URL = get_delineated_environment_variable('AVATAR_URL', 'https://usdos.sharepoint.com/_layouts/15/userphoto.aspx')
 TP_API_URL = get_delineated_environment_variable('TP_API_URL', 'http://mock_fsbid:3333/TrackingPrograms')
+AGENDA_ITEM_API_URL = get_delineated_environment_variable('AGENDA_ITEM_API_URL', 'http://mock_fsbid:3333/AgendaItems')
```
Comment (SidelineCory24) [inline talentmap_api/settings.py:545]: 🙌

File: talentmap_api/fsbid/urls/agenda.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+from django.conf.urls import url
+from rest_framework import routers
+
+from talentmap_api.fsbid.views import agenda as views
+
+router = routers.SimpleRouter()
+
+urlpatterns = [
+    url(r'^agenda_items/$', views.AgendaItemView.as_view(), name="agenda-agenda-items"),
```
Comment (SidelineCory24) [inline talentmap_api/fsbid/urls/agenda.py:9]: should it be `agenda-items` instead?

Comment (mjoyce91) [reply talentmap_api/fsbid/urls/agenda.py:9]: Looks like our other two-word endpoints use `_` opposed to `-`

Comment (SidelineCory24) [reply talentmap_api/fsbid/urls/agenda.py:9]: ah okay, I see it the other ones

File: talentmap_api/fsbid/services/common.py [hunk 3]
```diff
@@ -703,3 +703,28 @@ def sort_bids(bidlist, ordering_query):
...
+# in: in
+def convert_to_fsbid_ql(column = '', value = '', comparator = 'eq'):
+    if not column and not value and not comparator:
+        return None
+    return f"{column}|{comparator}|{value}"
+
+
+def categorize_remark(remark = ''):
+    obj = { 'title': remark, 'type': None }
+    if pydash.starts_with(remark, 'Creator') or pydash.starts_with(remark, 'CDO:') or pydash.starts_with(remark, 'Modifier'):
```
Comment (SidelineCory24) [inline talentmap_api/fsbid/services/common.py:717]: nice! 😎

File: talentmap_api/fsbid/services/common.py [hunk 4]
```diff
@@ -703,3 +703,28 @@ def sort_bids(bidlist, ordering_query):
...
+        obj['type'] = 'person'
+    return obj
+
+
+def parse_agenda_remarks(remarks = ''):
+    # split by semi colon
+    values = remarks.split(';')
+    # remove Nmn (no middle name) from Creator and CDO
+    values = pydash.map_(values, lambda o: pydash.reg_exp_replace(o, ' Nmn', '') if pydash.starts_with(o, 'Creator') or pydash.starts_with(o, 'CDO:') else o)
```
Comment (SidelineCory24) [inline talentmap_api/fsbid/services/common.py:726]: is this case sensitive for `Nmn`?

**Review by SidelineCory24 — APPROVED** [review]
Outside of `Nmn` question, looks good to go! 🚀 Will need to merge mock pr as well

File: talentmap_api/fsbid/services/common.py (hunk 4 above)
Comment (mjoyce91) [reply talentmap_api/fsbid/services/common.py:726]: Good point - in the real data it's always `Nmn`, but I like the idea to make it case insensitive - or at least add `NMN`. I'll update that

File: talentmap_api/fsbid/services/common.py [hunk 5]
```diff
@@ -703,3 +703,31 @@ def sort_bids(bidlist, ordering_query):
...
+
+
+def parse_agenda_remarks(remarks_string = ''):
+    remarks = remarks_string
+    if pydash.starts_with(remarks, 'Remarks:'):
+        remarks = pydash.reg_exp_replace(remarks_string, 'Remarks:', '', count=1)
+    # split by semi colon
+    values = remarks.split(';')
+    # remove Nmn (no middle name) from Creator and CDO
+    values = pydash.map_(values, lambda o: pydash.reg_exp_replace(o, ' Nmn', '', ignore_case=True) if pydash.starts_with(o, 'Creator') or pydash.starts_with(o, 'CDO:') else o)
```
Comment (SidelineCory24) [inline talentmap_api/fsbid/services/common.py:729]: 🚀

**Pull request merged**